unit_tests/src/test_ticket_administration.cpp

```diff
@@ -0,0 +1,365 @@
+
+        REQUIRE(write_coll_string == (general_query.begin().capture_results().at(0)));
+        REQUIRE(write_string == (general_query.begin().capture_results().at(1)));
+
+        tic::client::delete_ticket(conn, write_ticket_name);
+    }
+
+    SECTION("add user")
+    {
+        tic::client::create_ticket(conn, tic::ticket_type::read, set_ticket_collection, set_ticket_name);
+
+        tic::client::add_ticket_constraint(conn, set_ticket_name, tic::user_constraint{user_name});
+
+        std::string query = fmt::format("select TICKET_ALLOWED_USER_NAME where TICKET_STRING = '{}'", set_ticket_name);
+        auto general_query = builder.build<rcComm_t>(conn, query);
+
+        bool user_name_exists = false;
+
+        for (const auto& row: general_query) {
+            if (row[0] == user_name){
+                user_name_exists = true;
+                break;
+            }
+        }
+
+        REQUIRE(user_name_exists);
+
+        tic::client::delete_ticket(conn, set_ticket_name);
+    }
+
+    SECTION("remove user")
+    {
+        tic::client::create_ticket(conn, tic::ticket_type::read, set_ticket_collection, set_ticket_name);
+
+        tic::client::add_ticket_constraint(conn, set_ticket_name, tic::user_constraint{user_name});
+
+        tic::client::remove_ticket_constraint(conn, set_ticket_name, tic::user_constraint{user_name});
+
+        std::string query = fmt::format("select TICKET_ALLOWED_USER_NAME where TICKET_STRING = '{}'", set_ticket_name);
+        auto general_query = builder.build<rcComm_t>(conn, query);
+
+        REQUIRE(general_query.size() == 0);
+
+        tic::client::delete_ticket(conn, set_ticket_name);
+    }
+
+    SECTION("add group")
+    {
+        tic::client::create_ticket(conn, tic::ticket_type::write, set_ticket_collection, set_ticket_name);
+
+        tic::client::add_ticket_constraint(conn, set_ticket_name, tic::group_constraint{group_name});
+
+        std::string query = fmt::format("select TICKET_ALLOWED_GROUP_NAME where TICKET_STRING = '{}'", set_ticket_name);
+        auto general_query = builder.build<rcComm_t>(conn, query);
+
+        REQUIRE(general_query.size() > 0);
+        REQUIRE(group_name == (general_query.begin().capture_results().at(0)));
+
+        tic::client::delete_ticket(conn, set_ticket_name);
+    }
+
+    SECTION("remove group")
+    {
+        tic::client::create_ticket(conn, tic::ticket_type::write, set_ticket_collection, set_ticket_name);
+
+        tic::client::add_ticket_constraint(conn, set_ticket_name, tic::group_constraint{group_name});
+
+        tic::client::remove_ticket_constraint(conn, set_ticket_name, tic::group_constraint{group_name});
+
+        std::string query = fmt::format("select TICKET_ALLOWED_GROUP_NAME where TICKET_STRING = '{}'", set_ticket_name);
+        auto general_query = builder.build<rcComm_t>(conn, query);
+
+        REQUIRE(general_query.size() == 0);
+
+        tic::client::delete_ticket(conn, set_ticket_name);
+    }
```

Comment:
These tests can be combined. Can't test the remove operation without the add operation.